tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.hpp

```diff
@@ -132,14 +134,20 @@ struct make_eltwise_binary {
                     (in_a.get_legacy_shape() == in_b.get_legacy_shape()) or
                     (in_a.get_legacy_shape().without_padding() == in_b.get_legacy_shape().without_padding()),
                     "Input shapes must be the same!");
-                return operation::run(
+
+                DataType dtype = output_dtype.value_or(in_a.get_dtype());
```

Comment:
Will find a way to retrieve dtype in a more abstract way to not mention specific op type here.